Sources/NostrSDK/Events/Tags/ReplaceableEventCoordinates.swift

```diff
@@ -0,0 +1,118 @@
+//
+//  ReplaceableEventCoordinates.swift
+//
+//
+//  Created by Terry Yiu on 12/16/23.
+//
+
+import Foundation
+
+/// Coordinates to a (maybe parameterized) replaceable event.
+/// See [NIP-01 Tags](https://github.com/nostr-protocol/nips/blob/master/01.md#tags).
+public struct ReplaceableEventCoordinates: PubkeyProviding, RelayProviding, Equatable {
+    public static func == (lhs: Self, rhs: Self) -> Bool {
+        lhs.tag == rhs.tag
+    }
+
+    /// The tag representation of these replaceable event coordinates.
+    public let tag: Tag
+
+    private var tagValueSplit: [Substring] {
+        tag.value.split(separator: ":", omittingEmptySubsequences: false)
+    }
+
+    /// The kind integer of the referenced replaceable event.
+    /// Returns `nil` if the kind integer part of the tag value is malformed.
+    public var kind: EventKind? {
+        guard let kindInt = Int(String(tagValueSplit[0])) else {
```

Comment:
All direct array access like this should be protected against crashes like this:

If accessing `index`:
```swift
guard index < tagValueSplit.count else ...
```

Reply:
You're right. I thought that the initializer guard would be sufficient, but it's better to be safe to check here as well.